src/main/java/duke/command/Command.java

```diff
@@ -0,0 +1,204 @@
+        }
+        System.out.println("\tNice! I've marked this task as done:");
+        int index = taskNo - 1;
+        Task task = tasks.remove(index).doneTask();
+        System.out.println("\t" + task);
+        tasks.add(index, task);
+    }
+
+    /**
+     * Adds todos task to list of tasks.
+     *
+     * @param tasks  List of user's tasks.
+     * @param taskInfo  Description of todos task.
+     */
+    public void handleToDo(TaskList tasks, String taskInfo) throws DukeInvalidCommandException{
+        if (taskInfo.trim().equals("todo")) {
+            throw new DukeInvalidCommandException("The command is incomplete handsome :D");
+        }
+        taskInfo = taskInfo.replace("todo", "").trim();
+        tasks.addTask(new ToDos(taskInfo));
+    }
+
+    /**
+     * Adds deadline task to list of tasks.
+     *
+     * @param tasks  List of user's tasks.
+     * @param taskInfo  Description of deadline task.
+     */
+    public void handleDeadLine(TaskList tasks, String taskInfo) throws DukeInvalidCommandException, DukeDateTimeParseException{
+        taskInfo = taskInfo.replace("deadline", "");
+        String[] stringArr = taskInfo.split("/by", 2);
+        if (stringArr.length != 2) {
+            throw new DukeInvalidCommandException("The command is incomplete handsome :D");
+        }
+        taskInfo = stringArr[0].trim();
+        String by = stringArr[1].trim();
+        tasks.addTask(new Deadlines(taskInfo, by));
+    }
+
+    /**
+     * Adds event task to list of tasks.
+     *
+     * @param tasks  List of user's tasks.
+     * @param taskInfo  Description of event task.
+     */
+    public void handleEvent(TaskList tasks, String taskInfo) throws DukeInvalidCommandException, DukeDateTimeParseException{
+        taskInfo = taskInfo.replace("event", "");
+        String[] stringArr = taskInfo.split("/at", 2);
+        if (stringArr.length != 2) {
+            throw new DukeInvalidCommandException("The command is incomplete handsome :D");
+        }
+        taskInfo = stringArr[0].trim();
+        String at = stringArr[1].trim();
+        tasks.addTask(new Events(taskInfo, at));
+    }
+
+    /**
+     * Prints out the task if user input matches any word from task.
+     *
+     * @param tasks  List of user's tasks.
+     * @param taskInfo  Description of todos task.
+     * @return True if user input matches any word from any task in tasks list.
+     */
+    public boolean foundMatchingTasks(TaskList tasks, String taskInfo) {
+        boolean matched = false;
+        String[] taskInfos = taskInfo.trim().split(" ");
+        List<Task> matchList = tasks.returnMatchingTasks(taskInfos);
+        for(int i = 0; i < matchList.size(); i++) {
+            if (i == 0) {
+                System.out.println("\tHere are the matching tasks in your list:");
+            }
+            System.out.println(String.format("\t%d. %s", i + 1, matchList.get(i)));
+            matched = true;
+        }
+        return matched;

```

Comment:
The Command class should only handle common operations among commands, specific operations like matching can be put inside subclasses.

Reply:
I am following the format of week 3. Could you recommend me which subclass I should place it under? Thanks :D